TestLanguageConvertComponentSmoke/go failing in master

https://github.com/pulumi/pulumi/pull/13729 [failed](https://github.com/pulumi/pulumi/actions/runs/5874822298/job/15930362304#step:34:545) in master when running the tests with Go 1.19.x:

```
=== FAIL: . TestLanguageConvertComponentSmoke/go (0.48s)
    environment.go:97: Created new test environment:  /tmp/test-env2183890121
    smoke_test.go:174: Running command pulumi plugin ls
    smoke_test.go:179: Running command pulumi login --cloud-url file:///tmp/test-env2183890121
    smoke_test.go:180: Running command pulumi convert --language go --from pcl --out out
    smoke_test.go:180: Run Error: exit status 255
    smoke_test.go:180: STDOUT: Installing dependencies...
        
    smoke_test.go:180: STDERR: Converting from pcl...
        Converting to go...
        go: go.mod file indicates go 1.20, but maximum version supported by tidy is 1.19
        error: installing dependencies failed; rerun manually to try again, then run `pulumi up` to perform an initial deployment: `go mod tidy` failed to install dependencies: exit status 1
    smoke_test.go:180: Ran command pulumi args [convert --language go --from pcl --out out] and expected success. Instead got failure.
    --- FAIL: TestLanguageConvertComponentSmoke/go (0.48s)
```

We either need to:
1. Change to emitting 1.19 in the `go.mod` instead of 1.20
2. Move to testing on Go 1.21 and 1.20, since Go 1.19 is no longer supported

Likely (2) since we'll want to do that anyway.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -45,7 +45,7 @@ You'll want to install the following on your machine:
 You can get all required dependencies with brew and npm
 
 ```bash
-brew install node python@3 typescript yarn go@1.19 golangci/tap/golangci-lint gofumpt pulumi/tap/pulumictl coreutils jq
+brew install node python@3 typescript yarn go@1.21 golangci/tap/golangci-lint gofumpt pulumi/tap/pulumictl coreutils jq
 curl https://raw.githubusercontent.com/Homebrew/homebrew-cask/339862f79e/Casks/dotnet-sdk.rb > dotnet-sdk.rb
 brew install --HEAD -s dotnet-sdk.rb
 rm dotnet-sdk.rb

--- a/scripts/get-job-matrix.py
+++ b/scripts/get-job-matrix.py
@@ -115,7 +115,7 @@ ALL_PLATFORMS = ["ubuntu-latest", "windows-latest", "macos-latest"]
 MINIMUM_SUPPORTED_VERSION_SET = {
     "name": "minimum",
     "dotnet": "6",
-    "go": "1.19.x",
+    "go": "1.20.x",
     "nodejs": "16.x",
     "python": "3.9.x",
 }
@@ -123,7 +123,7 @@ MINIMUM_SUPPORTED_VERSION_SET = {
 CURRENT_VERSION_SET = {
     "name": "current",
     "dotnet": "7",
-    "go": "1.20.x",
+    "go": "1.21.x",
     "nodejs": "20.x",
     "python": "3.11.x",
 }

